### ISSUE TITLE
‘Instances’ column in training progress bar shows image batch size instead of instance count

### ISSUE THREAD

**Issue by Farenweh** [ultralytics/ultralytics#24737]
### Search before asking

- [x] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and found no similar bug report.

### Ultralytics YOLO Component

Train

### Bug

In detection training, the progress bar column is still labeled `Instances`, but the value being printed is the image batch size of the current rank, not the number of object instances (It used to be but was changed during commit 93e0555251bd7eef5ac8fdf367200d32a5967053).

In current upstream `main` at commit `3ca0b4fc373c01522da1a6ec25710516ae21beb2`:

```python
# ultralytics/models/yolo/detect/train.py
"Instances",
```

But the actual value logged in the progress bar is:

```python
# ultralytics/engine/trainer.py
batch["img"].shape[0],  # batch size, i.e. 8
```

This is misleading, especially in DDP training. For example, with global `batch=1024` over 16 DDP processes, the progress bar always shows `Instances=64`, but `64` is the per-rank image batch size, not the number of ground-truth object instances.

### Environment

This issue is not hardware-specific. It is reproducible by source inspection on upstream `main`:

```text
Ultralytics YOLO commit: 3ca0b4fc373c01522da1a6ec25710516ae21beb2
Component: Train
Task: detect
Mode: train
```

The mismatch exists in the upstream source:

```python
# Header:
# ultralytics/models/yolo/detect/train.py
"Instances"

# Logged value:
# ultralytics/engine/trainer.py
batch["img"].shape[0]
```

Observed during distributed detection training, but the bug is independent of OS, CUDA/NPU, or DDP. DDP only makes it more obvious because the displayed value becomes the per-rank image batch size.

### Minimal Reproducible Example

```python
from ultralytics import YOLO

model = YOLO("yolo11n.yaml")

model.train(
    data="coco8.yaml",
    imgsz=640,
    epochs=1,
    batch=4,
)
```

### Additional

This appears to have been introduced in:

```text
93e0555251bd7eef5ac8fdf367200d32a5967053
`ultralytics 8.4.52` YOLO Semantic Segmentation models (#24518)
```

Relevant diff:

```diff
- batch["cls"].shape[0],  # batch size, i.e. 8
+ batch["img"].shape[0],  # batch size, i.e. 8
```

The value changed from label/object count to image batch size, but the detection trainer progress header still says `Instances`.

### Are you willing to submit a PR?

- [ ] Yes I'd like to help by submitting a PR!

**Comment by UltralyticsAssistant** [comment]
👋 Hello @Farenweh, thank you for the clear bug report and for including a reproducible example 🚀. This is an automated response to help triage the issue, and an Ultralytics engineer will also assist you soon.

We appreciate the detailed source references, commit information, and explanation of why the `Instances` value is misleading in training output 🔍. Since you already provided a [minimum reproducible example](https://docs.ultralytics.com/help/minimum-reproducible-example/), that helps a lot with investigation.

If this is a custom training ❓ Question, please provide as much information as possible, including dataset image examples and training logs, and verify you are following our [Tips for Best Training Results](https://docs.ultralytics.com/guides/model-training-tips).

Join the Ultralytics community where it suits you best. For real-time chat, head to [Discord](https://discord.com/invite/ultralytics) 🎧. Prefer in-depth discussions? Check out [Discourse](https://community.ultralytics.com/). Or dive into threads on our [Subreddit](https://www.reddit.com/r/Ultralytics/) to share knowledge with the community.

## Upgrade

Upgrade to the latest `ultralytics` package including all [requirements](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml) in a [**Python>=3.8**](https://www.python.org/) environment with [**PyTorch>=1.8**](https://pytorch.org/get-started/locally/) to verify your issue is not already resolved in the latest version:

```bash
pip install -U ultralytics
```

## Environments

YOLO may be run in any of the following up-to-date verified environments (with all dependencies including [CUDA](https://developer.nvidia.com/cuda)/[CUDNN](https://developer.nvidia.com/cudnn), [Python](https://www.python.org/) and [PyTorch](https://pytorch.org/) preinstalled):

- **Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo26"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VM. See [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial)
- **Amazon** Deep Learning AMI. See [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial)
- **Docker Image**. See [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml/badge.svg" alt="Ultralytics CI"></a>

If this badge is green, all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule) tests are currently passing. CI tests verify correct operation of all YOLO [Modes](https://docs.ultralytics.com/modes) and [Tasks](https://docs.ultralytics.com/tasks) on macOS, Windows, and Ubuntu every 24 hours and on every commit.

**Comment by Y-T-G** [comment]
Hello @Farenweh 

Thanks for reporting. Can you test with 

```
pip install git+https://github.com/ultralytics/ultralytics@instance-count-fix
```

**Comment by Farenweh** [comment]
I cherry-picked commits `ab9db264dc996fbc84ed1112e39275cb6b177f76` and `b157ae74cc40c98387af95a9f597572577e25a2f` from the `instance-count-fix` branch onto my own branch and tested them. Everything works correctly on my side now.

<img width="1806" height="90" alt="Image" src="https://github.com/user-attachments/assets/b79a5c21-7a98-4d6e-ab7b-7d0eac5c44b5" />

Thanks for the quick fix!

**Comment by UltralyticsAssistant** [comment]
A potential fix for this issue has now been merged in [PR #24738: Fix instances count logging](https://github.com/ultralytics/ultralytics/pull/24738).

This change updates training progress reporting in `ultralytics/engine/trainer.py` so the logged count better reflects what is actually being processed:

- Previously, the log always used the image batch size
- It now uses `batch.get("cls", batch["img"]).shape[0]`
- For classification tasks, this means the count comes from the `cls` tensor when available
- For other tasks, it safely falls back to the image batch size
- The related label/comment was also updated from “batch size” to “no. of instances” to better match the displayed value

This is a low-risk change that only affects logging and display, not training behavior or results.

Credit to @Y-T-G for the PR, with contributions from @onuralpszr and @UltralyticsAssistant.

If you'd like to test whether this resolves the issue, you can install the latest `main` branch:

```bash
pip install git+https://github.com/ultralytics/ultralytics.git@main
```

Or, if preferred, wait for the next official PyPI release.

If possible, please try the updated code and let us know whether it resolves the issue on your side. Thanks 🙏 for reporting this, and any further feedback is welcome if the issue still persists.
